### PR TITLE
Add logic for AT90USBxx7 where needed

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Setting `MCU = at90usb647` or `1287` would fail to compile, because there were no checks in mcu_selection.mk, bootloader.mk, and a few other places, that would catch it.
The xx7 parts seem to be essentially identical to the xx6, however they [also have the ability to act as a USB host](http://www.fourwalledcubicle.com/files/LUFA/Doc/170418/html/_page__a_v_r8_support.html). I don't think you can be in both modes at the same time though, but QMK doesn't have any host-mode capability anyway.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* https://old.reddit.com/r/olkb/comments/ii66i5/does_qmk_support_at90usb1287/

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]

**Comment by drashna** [comment]
Issue here. if you set the MCU to the xxx7 variant, it won't compile because of the `RESET` keycode.  Removing this keycode fixes the issue.   

Specifically, thi issue is in <avr/iousbxx6_7.h>, line 733 (at least for 8.4.0.  The at90usbxxx6 has `RESET` defined as part of it's config. Likely for how the USB is actually handled.

File: quantum/keymap.h [hunk 1]
```diff
@@ -40,6 +40,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #if defined(PROTOCOL_CHIBIOS)
 #    define RESET QK_RESET
 #endif
+// Gross hack, remove me and change RESET keycode to QK_BOOT
+#if defined(__AVR_AT90USB647__) || defined(__AVR_AT90USB1287__)
+#    undef RESET
+#endif
```
Comment (fauxpark) [inline quantum/keymap.h:46]: @drashna what do you think about this? :P
`donutcables/scrabblepad:default` now compiles with MCU set to 1287.

Comment (drashna) [reply quantum/keymap.h:46]: I'm really not a fan of this. but short of this or something similar, for now ...  I'm not sure what else we can really do.

Comment (fauxpark) [reply quantum/keymap.h:46]: Yeah, but at least it's right next to a similar hack for ChibiOS, so when we eventually get round to renaming `RESET`, we can take them both out in one fell swoop.

**Pull request merged**